design-document.md

```diff
@@ -106,6 +123,167 @@ to achieve these goals:
+       clients do besides uploading data shares and accordingly, such attacks
+       are outside of the threat model.
+1. Clients may affect the quality of aggregations by reporting false data.
+     * Prio can only prove that submitted data is valid, not that it is true.
+       False data can be mitigated orthogonally to the Prio protocol (e.g., by
+       requiring that aggregations include a minimum number of contributions)
+       and so these attacks are considered to be outside of the threat model.
+1. Clients can send invalid encodings of data.
+
+##### Mitigations
+
+1. The shared validity proof evaluated by aggregators prevents either individual
+   clients or coalitions of clients from compromising the robustness property.
+
+#### Aggregator
+
+##### Assets
+
+1. Unencrypted data shares.
+1. Data share decryption keys.
+1. Client identifying information.
+1. Aggregation parts.
+1. Aggregator identity.
+
+##### Capabilities
+
+1. Aggregators may defeat the robustness of the system by emitting bogus
+   aggregation shares.
+1. If clients reveal identifying information to aggregators (such as a trusted
+   identity during client authentication), aggregators can learn which clients
+   are contributing data.
+     1. Aggregators may weaken anonymity by revealing that a particular client
+        contributed data to the system.
+     1. Aggregators may choose to selectively omit data from certain clients.
+          * For example, omitting submissions from a particular geographic
+            region to falsely suggest that a particular localization is not
+            being used.
+1. Individual aggregators may compromise availability of the system by refusing
+   to emit aggregation parts.
+
+##### Mitigations
+
+1. The linear secret sharing scheme employed by the client ensures that privacy
+   is preserved unless every participating aggregator reveals their unencrypted
+   data shares.
+1. Infrastructure diversity. Prio deployments should ensure that aggregators do
+   not have common dependencies that would enable a single vendor to reassemble
+   data.
+     * For example, if all participating aggregators stored unencrypted data
+       shares on the same cloud object storage service, then that cloud vendor
+       would be able to reassemble all the data shares and defeat privacy.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and accepting any result
+   would prevent an individual aggregator from compromising availability.
+1. Running the protocol over multiple subsets of the available aggregators
+   chosen so that no aggregator appears in all subsets and requiring that the
+   results agree would prevent an individual aggregator from compromising
+   robustness
+     * This could also allow identifying of defective aggregators, by finding
+       the aggregator that appears in every subset whose results were wrong.
```

Comment:
^ Does this defeat the bogus aggregation share attack? I'm not sure anything does. There may be other, more subtle attacks it detects, but given how easy it is for an aggregator to submit a bogus aggregation share, I don't think it matters much. (See comment in "Future work and possible extensions".)

However, detecting bad aggregators is a really good idea. We'll need to flesh this out later.